File: src/main/java/keri/projectx/machine/gui/ContainerFabricator.scala
//package keri.projectx.machine.gui
//
//import codechicken.lib.inventory.container.{ContainerExtended, SlotDummy, SlotDummyOutput}
//import codechicken.lib.packet.PacketCustom
//import keri.projectx.machine.tile.TileFabricator
//import com.projectxy.world.ProjectXYWorld
//import net.minecraft.entity.player.{EntityPlayer, InventoryPlayer}
//import net.minecraft.inventory.{ClickType, Slot}
//import net.minecraft.item.ItemStack
//import net.minecraftforge.fml.relauncher.{Side, SideOnly}
//
//class ContainerFabricator(inventoryPlayer: InventoryPlayer, tile: TileFabricator) extends ContainerExtended {
//  for (row <- 0 until 3)
//    for (column <- 0 until 3)
//      addSlotToContainer(new SlotDummy(tile, row * 3 + column, 8 + column * 18, 30 + row * 18, 1) {
//        override def canTakeStack(player: EntityPlayer): Boolean = false
//      })
//  addSlotToContainer(new SlotDummyOutput(tile, 9, 79, 48) {
//    override def slotClick(container: ContainerExtended, player: EntityPlayer, button: Int, modifier: ClickType): ItemStack = {
//      //Clears crafting in one click
//      if (button == 1)
//        for (i <- 0 until 9)
//          tile.setInventorySlotContents(i, ItemStack.EMPTY)
//      ItemStack.EMPTY
//    }
//
//    override def canTakeStack(player: EntityPlayer): Boolean = false
//  })
//  for (row <- 0 until 3)
//    for (column <- 0 until 3)
//      addSlotToContainer(new Slot(tile, row * 3 + column + 10, 116 + column * 18, 30 + row * 18))
//  bindPlayerInventory(inventoryPlayer, 8, 93)
//
//  val REDSTONE_SIGNAL_CHANGE = 1
//  val CRAFTING_SLOTS_CHANGED = 2
//
//  override def doMergeStackAreas(slotIndex: Int, stack: ItemStack): Boolean = {
//    if (slotIndex < 19) return mergeItemStack(stack, 19, 55, true)
//    mergeItemStack(stack, 10, 19, false)
//  }
//
//  @SideOnly(Side.CLIENT)
//  def handleGuiChange(selected_redstone_signal: Int): Unit = {
//    tile.mode = selected_redstone_signal
//    tile.markDirty()
//    val packet = getPacket()
//    packet.writeInt(REDSTONE_SIGNAL_CHANGE)
//    packet.writeInt(selected_redstone_signal)
//    packet.sendToServer()
//  }
//
//  def getPacket(): PacketCustom = new PacketCustom(ProjectXYWorld.MOD_ID, 1)
//
//  def changeCraftingRecipe(stacks: Array[ItemStack]): Unit = {
//    val packet = getPacket()
//    packet.writeInt(CRAFTING_SLOTS_CHANGED)
//    for (i <- stacks.indices) {
//      packet.writeItemStack(stacks(i))
//    }
//    packet.sendToServer()
//  }
//
//  def handleServerPacket(packet: PacketCustom): Unit = {
//    packet.readInt match {
//      case REDSTONE_SIGNAL_CHANGE =>
//        tile.mode = packet.readInt
//        tile.markDirty()
//      case CRAFTING_SLOTS_CHANGED =>
//        for (i <- 0 until 9)
//          tile.setInventorySlotContents(i, packet.readItemStack())
//      case _ =>
//    }
//  }
//}

File: src/main/scala/keri/projectx/container/ContainerFabricator.java
/*
 * Copyright (c) 2017 KitsuneAlex & Adam8234. All rights reserved!
 * Do not destribute or redistribute this software without the
 * explicit permission of the developer!
 */

package keri.projectx.container;

import codechicken.lib.inventory.container.ContainerExtended;
import codechicken.lib.inventory.container.SlotDummy;
import codechicken.lib.inventory.container.SlotDummyOutput;
import keri.projectx.tile.TileEntityFabricator;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.inventory.ClickType;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;

import javax.annotation.Nonnull;

public class ContainerFabricator extends ContainerExtended {

    private InventoryPlayer inventoryPlayer;
    private TileEntityFabricator tile;

    public ContainerFabricator(InventoryPlayer inventoryPlayer, TileEntityFabricator tile) {
        this.inventoryPlayer = inventoryPlayer;
        this.tile = tile;
        this.bindPlayerInventory(inventoryPlayer, 8, 93);

        for (int x = 0; x < 3; x++) {
            for (int y = 0; y < 3; y++) {
                int index = x + 3 * y;
                int posX = 8 + 18 * x;
                int posY = 30 + 18 * y;
                this.addSlotToContainer(new SlotDummy(tile, index, posX, posY) {
                    @Override
                    public void putStack(@Nonnull ItemStack stack) {
                        if (!stack.isEmpty()) {
                            stack.setCount(1);
                        }
                        super.putStack(stack);
                    }
                });
            }
        }

        this.addSlotToContainer(new SlotDummyOutput(tile, 9, 79, 48) {
            @Override
            public ItemStack slotClick(ContainerExtended container, EntityPlayer player, int button, ClickType clickType) {
                //Clears crafting in one click
                if (button == 1)
                    for (int i = 0; i < 9; i++)
                        tile.setInventorySlotContents(i, ItemStack.EMPTY);
                return ItemStack.EMPTY;
            }
        });

        for (int x = 0; x < 3; x++) {
            for (int y = 0; y < 3; y++) {
                int index = (x + 3 * y) + 10;
                int posX = 116 + 18 * x;
                int posY = 30 + 18 * y;
                this.addSlotToContainer(new Slot(tile, index, posX, posY));
            }
        }
    }

    @Override
    public boolean canInteractWith(EntityPlayer player) {
        return this.tile.isUsableByPlayer(player);
    }

}
